app/src/main/java/com/google/droidjump/MainActivity.java

```diff
@@ -150,6 +175,29 @@ protected void onResume() {
         }
     }
 
+    public Task<AnnotatedData<PlayerBuffer>> loadFriendNames() {
+        return getPlayersClient().loadFriends(GameConstants.FRIENDS_PER_PAGE, /* forceReload = */ false)
+                .addOnSuccessListener(buffer -> {
+                    playerBuffer = buffer.get();
+                    try {
+                        while (DataBufferUtils.hasNextPage(playerBuffer)) {
+                            getPlayersClient().loadMoreFriends(GameConstants.FRIENDS_PER_PAGE).addOnSuccessListener(data -> {
+                                playerBuffer = data.get();
+                            });
+                        }
+                    } finally {
+                        friendNames.clear();
+                        for (Player player : playerBuffer) {
+                            String friendName = player.freeze().getDisplayName();
+                            friendNames.add(friendName);
```

Comment:
how is the order of the friends name preserved?

Reply:
Alphabetically.